src/FilenameParsing/AlternativeFileExtensionTrait.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace SilverStripe\Assets\FilenameParsing;
+
+use SilverStripe\Core\Convert;
+
+/**
+ * This trait can be applied to a `FileIDHelper` imlementation to add the baility to encode alternative file extension.
+ *
+ * This allows the creation of file variant with different extension to the original file.
+ */
+trait AlternativeFileExtensionTrait
```

Comment:
I'm thinking maybe this should be some sort of service instead. There's not really any value in making that logic snipet part of the FIleIDHelper.

Also, there's bit of logic in `ImageManipulate::manipulateExtension()` and `InterventionBackend` that might make more sense in a centralised service.